File: files/CH-081.py
import pandas as pd

path = "CH-081 Compaint grouping.xlsx"

input1 = pd.read_excel(path, skiprows=1, usecols="B:F", nrows= 4)
input2 = pd.read_excel(path, skiprows=1, usecols="H:J")
test = pd.read_excel(path, skiprows=1, usecols="K:K")

result1 = input1.melt(id_vars=["From-To"], var_name="To", value_name="Distance")

result2 = input2.assign(Path=input2["Path"].str.split(",")).explode("Path")
result2["to"] = result2.groupby(["Date","Staff ID"])["Path"].shift(-1)
result2 = result2.dropna()

result = result2.merge(result1, left_on=["Path", "to"], right_on=["From-To", "To"])
result = result.groupby(["Date", "Staff ID"]).agg({"Distance": "sum"}).reset_index()

print(result["Distance"].equals(test["Distance"])) # True

File: files/000-099/081/CH-081.R
library(tidyverse)
library(readxl)


path = "files/CH-081 Compaint grouping.xlsx"
input1 = read_xlsx(path, range = "B2:F6")
input2 = read_xlsx(path, range = "H2:J14")
test  = read_xlsx(path, range = "K2:K14")

result1 = input1 %>%
  pivot_longer(cols = -c(`From-To`), names_to = "To", values_to = "Distance")

result2 = input2 %>%
  separate_rows(Path, sep = ",") %>%
  mutate(to = lead(Path), .by = c(Date, `Staff ID`)) %>%
  na.omit()

result = result2 %>%
  left_join(result1, by = c("Path" = "From-To", "to" = "To")) %>%
  summarise(Distance = sum(Distance, na.rm = TRUE), .by = c(Date, `Staff ID`))

identical(result$Distance, test$Distance)
#> [1] TRUE